src/games/strategy/engine/lobby/client/ui/LobbyGameTableModel.java

```diff
@@ -121,10 +123,18 @@ private void updateGame(final GUID gameId, final GameDescription description) {
     SwingUtilities.invokeLater(new Runnable() {
       @Override
       public void run() {
-        Tuple<GUID,GameDescription> toReplace = findGame(gameId);
-        int replaceIndex = gameList.indexOf(toReplace);
-        gameList.set(replaceIndex, Tuple.of(gameId, description));
-        fireTableRowsUpdated(replaceIndex, replaceIndex);
+        if (gameId == null) {
+          return;
+        }
+
+        final Tuple<GUID, GameDescription> toReplace = findGame(gameId);
+        if (toReplace == null) {
+          gameList.add(Tuple.of(gameId, description));
```

Comment:
The call notifies listeners and we never add any, it's an unused feature, dead code. If it were used, we modify the table model in other places without firing events, so it would be buggy if it were ever used. Having a placeholder hook for an unused feature is over engineering.


First we can see we never add any listeners by tracing down usages, there is only one of LobbyGameTableModel:
```
    m_gameTableModel = new LobbyGameTableModel(m_messengers.getMessenger(), m_messengers.getChannelMessenger(),
        m_messengers.getRemoteMessenger());
```
Then there are three usages of that variable:
```
    final GameDescription description = m_gameTableModel.get(modelIndex);
    final GameDescription description = m_gameTableModel.get(modelIndex);
    final GameDescription description = m_gameTableModel.get(modelIndex);
```
So we can see we never add listeners.

Now to double check that the method doesn't do any UI repaint or refresh work, we can look at that source code:


```
    public void fireTableRowsUpdated(int firstRow, int lastRow) {
        fireTableChanged(new TableModelEvent(this, firstRow, lastRow,
                             TableModelEvent.ALL_COLUMNS, TableModelEvent.UPDATE));
    }


    public void fireTableChanged(TableModelEvent e) {
        // Guaranteed to return a non-null array
        Object[] listeners = listenerList.getListenerList();
        // Process the listeners last to first, notifying
        // those that are interested in this event
        for (int i = listeners.length-2; i>=0; i-=2) {
            if (listeners[i]==TableModelListener.class) {
                ((TableModelListener)listeners[i+1]).tableChanged(e);
            }
        }
    }
```

So all it does is really just notify listeners, and we never add any.